pallets/subtensor/tests/epoch.rs

```diff
@@ -583,6 +583,298 @@ fn test_16384_graph_sparse() {
 	});
 }
 
+// Test bonds exponential moving average over a sequence of epochs.
+#[test]
+fn test_bonds() {
+	new_test_ext().execute_with(|| {
+		let sparse: bool = true;
+		let n: u16 = 8;
```

Comment:
What is `n`?
> I see it being used as a limit for the `for` loop, and as a value for `max_allowed_validators` related stuff too

Reply:
In these tests it typically refers to the size of the network.